src/fights.ts

```diff
@@ -716,7 +721,7 @@ const freeFightSources = [
     }
   ),
   new FreeFight(
-    () => (have($item`molehill mountain`) && !get("_molehillMountainUsed") ? 1 : 0),
+    () => (molemanReady() && !thesisReady() ? 1 : 0),
```

Comment:
Based on the way thesisReady works, people who don't own a prof will never fight the moleman.

Reply:
You're quite right. I've adjusted the logic.

Reply:
Hang on, isn't `(thesisReady() ? 0 : molemanReady() ? 1 : 0)` the same as `(molemanReady() && !thesisReady() ? 1 : 0)`? How will people without a Prof be blocked from the fight?
| Thesis | Moleman  | desired |
|--------|-------|---------|
| TRUE   | TRUE  | 0       |
| TRUE   | FALSE | 0       |
| FALSE  | TRUE  | 1       |
| FALSE  | FALSE | 0       |

Reply:
oh, I just straight-up misread that then. You're right!

Reply:
Hehe cool. The refactored version probably reads a bit easier. Thanks for the approve.